src/hb-ot-name.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2018  Ebrahim Byagowi
+ *
+ *  This is part of HarfBuzz, a text shaping library.
+ *
+ * Permission is hereby granted, without written agreement and without
+ * license or royalty fees, to use, copy, modify, and distribute this
+ * software and its documentation for any purpose, provided that the
+ * above copyright notice and the following two paragraphs appear in
+ * all copies of this software.
+ *
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE TO ANY PARTY FOR
+ * DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES
+ * ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN
+ * IF THE COPYRIGHT HOLDER HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * THE COPYRIGHT HOLDER SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING,
+ * BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE.  THE SOFTWARE PROVIDED HEREUNDER IS
+ * ON AN "AS IS" BASIS, AND THE COPYRIGHT HOLDER HAS NO OBLIGATION TO
+ * PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+#include "hb.hh"
+
+#include "hb-ot-face.hh"
+#include "hb-ot-name-table.hh"
+
+static inline const OT::name& _get_name (hb_face_t *face)
+{
+  if (unlikely (!hb_ot_shaper_face_data_ensure (face))) return Null(OT::name);
+  return *hb_ot_face_data (face)->name;
+}
+
+/**
+ * hb_ot_name_entries_count:
+ * @face: #hb_face_t
+ * @name_index:
+ * @start_offset:
+ * @languages_length: (in/out)
+ * @languages: (out)
+ *
+ * Return value:
+ *
+ * Since: REPLACEME
+ **/
+unsigned int
+hb_ot_name_get_languages (hb_face_t    *face,
+			  unsigned int  name_index,
+			  unsigned int  start_offset,
+			  unsigned int *languages_length,
+			  unsigned int *languages)
+{
+  const OT::name& table = _get_name (face);
+
+  unsigned int matched = 0;
+  for (unsigned int i = 0; i < table.count; i++)
```

Comment:
Sad about linear search but other than computers are fast nowadays, not sure how to fill the languages buffer correctly.

Reply:
I thought your agreement on the other thread was because of agreeing with the API Behdad was suggesting :P

Reply:
Yeah thinking about it more, since this is going to be really slow, let's leave it out now.

Reply:
and instead provide the lower-level items so advanced clients (eg. fontconfig) can figure it out themselves.